### PR TITLE
Build godot project on CI

### PR DESCRIPTION
**Pull request by Dreia**
This workflow outputs an Island.zip file into_ build/mods_ folder (if successful). It is meant to run on workflow dispatch, that means it creates a run button.

### :ballot_box_with_check: Definition of Done checklist
- [ ] Meaningful title and valuable description in pull request
- [ ] Code change tested manually if not covered by automated tests
- [ ] Pull request linked to one or more issues/stories
- [ ] Acceptance criteria of the linked stories satisfied
- [ ] Code change comprehended by those who approved it
- [ ] State of target branch maintained or improved after the code change

### CODE REVIEW

File: .github/workflows/Godot CI.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          lfs: true
+      - name: Setup
+        run: |
+          mkdir -v -p ~/.local/share/godot/templates
+          mv /root/.local/share/godot/templates/${GODOT_VERSION}.stable ~/.local/share/godot/templates/${GODOT_VERSION}.stable
+      - name: Resolve Dependencies
+        run: |
+          mkdir -v -p build/linux
+          cd $EXPORT_NAME
+          timeout 30s godot . --editor --no-window
```
Comment (RuthgerD) [inline .github/workflows/Godot CI.yml:29]: if this works then you do not have to move the templates at all FYI as they are unused when just zipping

Comment (Dreia) [reply .github/workflows/Godot CI.yml:29]: Thanks, I'll remove that step.

**Comment by RuthgerD** [comment]
I suggest trying this as well instead of doing the timeout thing:
```
godot --export-pack linux --no-window output.pck
```
(with `linux` being the export preset you have, I think you have "Linux/X11")

**Comment by Dreia** [comment]
It looks like it does nothing although the checks pass. If I export then I need to move the templates as well right ? Do I also need to get the headless version of Godot, because I think for that I really need to download the binary and set it up on PATH like you guys did.

File: .github/workflows/Godot CI.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+name: Build Island.zip Artifact
+
+on: [workflow_dispatch]
```
Comment (RuthgerD) [inline .github/workflows/Godot CI.yml:3]: This needs to be how it was before i.e [push, pull_request]

**Comment by RuthgerD** [comment]
timeout is probably returning a non 0 exit code which is fine but the action is thinking it failed

**Comment by Dreia** [comment]
I am not sure if i should see a build directory being created with the zip file in it ?

**Comment by RuthgerD** [comment]
image
Looks like the mod works! nice job :+1:

**Comment by Dreia** [comment]
Thanks, should I merge then ? I can't find the file tbh so I can't check the name

File: .github/workflows/Godot CI.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+          lfs: true
+      - name: Resolve Dependencies
+        run: |
+          mkdir -v -p build/mods
+          cd $DIR
+          godot --export-pack "Linux/X11" --no-window ../build/mods/$EXPORT_NAME.pck
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v1
+        with: # the final result should be called Island.zip
+          name: $EXPORT_NAME
```
Comment (RuthgerD) [inline .github/workflows/Godot CI.yml:29]: variable expansion works different in github actions (opposed to bash) so it would be `${{ EXPORT_NAME }}`

**Comment by RuthgerD** [comment]
>  I can't find the file tbh so I can't check the name
If you want to find the artifacts, click on the little green checkmark next to the commit and go to the details of the respective action, from there go to "Summary" and it should be easy from there.

**Review by RuthgerD — APPROVED** [review]
Seems like it does what we want now, the action yml itself can be improved but not really important

**Pull request merged**